### PR TITLE
Remove legacy providers interface

### LINKED ISSUE #375: Remove legacy providers interface
### What is the expected enhancement?

Remove the legacy providers interface, especially `BaseBackend` that is referenced in `QuantumInstance` type hints.
See https://github.com/Qiskit/qiskit-terra/pull/7886 for more details.

### PR DESCRIPTION
**Pull request by manoelmarques**
### Summary
FIxes #375

### CODE REVIEW

**Comment by adekusar-drl** [comment]
I agree with @woodsp-ibm to keep this on hold until the things get merged in Terra and we release QML 0.4.

**Comment by manoelmarques** [comment]
Terra removed the legacy interfaces from its code and now the build fails. This PR needs to be merged.

**Comment by adekusar-drl** [comment]
> Terra removed the legacy interfaces from its code and now the build fails. This PR needs to be merged.

We were waiting too long :)

**Review by adekusar-drl — APPROVED** [review]

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**
